app/src/main/java/com/zeapo/pwdstore/git/operation/GitOperation.kt

```diff
@@ -55,6 +56,7 @@ abstract class GitOperation(protected val callingActivity: FragmentActivity) {
     protected val repository = PasswordRepository.getRepository(null)!!
     protected val git = Git(repository)
     protected val remoteBranch = GitSettings.branch
+    private val authActivity get() = callingActivity as ContinuationContainerActivity
```

Comment:
`GitOperation` also executes our committing flow which does not invoke authentication related methods, so we get around the overhead of changing every class to extend the container by typecasting at runtime. We've ensured that git operations can only be called from a select few entrypoints which are regulated by us, so this is 'safe'.